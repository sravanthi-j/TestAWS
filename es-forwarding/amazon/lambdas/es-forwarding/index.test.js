const index = require('./index');
const expect = require('expect');
const zlib = require("zlib");
const { processMessage } = require('./esforwarding/es-forwarding');

test('Wrapper has the right context', ()=> {

    var obj = {
        processMessage: function(something, callback) {

            expect(this.testProp).toBe(5);

        },
        testProp: 5
        
    }

    var wrapped = index._wrap(obj,obj.processMessage);
    wrapped({}, {
        succeed: ()=>{}
    });

});

test('Unzips the buffer and handles message', (callback) => {
    
    var counters = {invokeHandlers: 0};

    index.invokeHandlers = (b, context)=>{
        expect(b).toMatchSnapshot("noncontrol");
        
        ++ counters.invokeHandlers; context.succeed();
    };

    var content = {
        messageType: "", 
        awsLogs: []

    };

    zlib.gzip(JSON.stringify(content), function(err, result){
 
       
        expect(err).toBeFalsy();
     
        index.handler({awslogs: {data: result.toString('base64')}}, {succeed: function(){

  
            expect(counters.invokeHandlers).toBe(1);
            callback();
            
        }});  

    });

});

 

test('Unzips the buffer and handles control message', (callback) => {
    
    var counters = {invokeHandlers: 0};

    index.invokeHandlers = (b, context)=>{
         ++ counters.invokeHandlers;
        context.succeed();
    };
    var content = {
        messageType: "CONTROL_MESSAGE", 
        awsLogs: []

    };

    zlib.gzip(JSON.stringify(content), function(err, result){
        expect(err).toBeFalsy();

        index.handler({awslogs: {data: result.toString('base64')}}, {succeed: function(){



        
            expect(counters.invokeHandlers).toBe(0);
            callback();
    

        }});

    });

});

 