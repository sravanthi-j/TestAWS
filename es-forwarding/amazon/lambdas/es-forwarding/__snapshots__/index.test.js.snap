// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unzips the buffer and handles message: noncontrol 1`] = `
Object {
  "awsLogs": Array [],
  "messageType": "",
}
`;
